src/Orchard.Web/Modules/Orchard.Users/Annotations/UsernameValidLengthAttribute.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Orchard.Localization;
+using Orchard.Users.Models;
+
+namespace Orchard.Users.Annotations {
```

Comment:
You can delete it. It's ok to not validate this, and the way it's made as an attribute doesn't make sense since it's not reusable, it's checking explicitly which properties it's applied to. It's less ugly to just remove the check.

Reply:
I removed the file and changed the service so if Maximum username length and Minimum username length values don't make sense those settings are ignored during the validation.